### PR TITLE
Added option to copy the DOI to the copy sub-menu

### LINKED ISSUE #7826: Add posibility to just copy the DOI
**Is your suggestion for improvement related to a problem? Please describe.**
I'm currently writing a review for a paper, where I usually provide DOIs to facilitate authors retrieving the references. To copy the DOI I need to open the respective entry, navigate to the correct tab and then copy it. 

**Describe the solution you'd like**
It would be helpful to include a further entry in the copy menu (right click + shortcuts) to quickly copy the DOI.

### PR DESCRIPTION
**Pull request by AidanM11**
Adds the feature requested in 
Fixes #7826 
- Modelled off already existing "Copy title" and "Copy key" features
- Adds method copyDOI to CopyMoreAction and binds it appropriately
- Adds new item in RightClickMenu
- Adds appropriate tests to CopyMoreActionTests
- Adds appropriate localization entries
Screenshot:
CopyDOIButton

.

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by calixtus** [comment]
Thank you for your interest in this issue, especially for thinking of the tests.
Please do not forget to check the other checkboxes in the pr description too.

**Comment by AidanM11** [comment]
I've completed the checklist, and everything seems right. The only problem is the checkstyle test failing, saying that there are 5 trailing spaces in CHANGELOG.md, but my change doesn't appear to add any trailing spaces. I'm not sure what to do to fix that.

File: src/test/java/org/jabref/gui/edit/CopyMoreActionTest.java [hunk 1]
```diff
@@ -40,6 +40,7 @@
     private BibEntry entry;
     private List<String> titles = new ArrayList<String>();
     private List<String> keys = new ArrayList<String>();
+    private List<String> DOIs = new ArrayList<String>();
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/gui/edit/CopyMoreActionTest.java:43]: Variabl names  should be all lowerCamelCase, e.g dois

File: src/main/java/org/jabref/gui/edit/CopyMoreAction.java [hunk 2]
```diff
@@ -113,6 +114,33 @@ private void copyKey() {
         }
     }
 
+    private void copyDOI() {
+        List<BibEntry> entries = stateManager.getSelectedEntries();
+
+        // Collect all non-null DOIs.
+        List<String> DOIs = entries.stream()
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/edit/CopyMoreAction.java:121]: variables should be all lower camel case, event is they are abbreviations => doi

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
Overall looks good, just one minor nitpicking.

File: src/main/java/org/jabref/gui/edit/CopyMoreAction.java (hunk 2 above)
Comment (AidanM11) [reply src/main/java/org/jabref/gui/edit/CopyMoreAction.java]: This should be addressed by commit 595945eb3c918f330f00d4b604382fe32fbb1f66. Thanks for the feedback.

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks for the quick follow up. Looks good from my side. 
PS: For the next time, it's better to use a separate branch instead of main to avoid unnecessary conflicts. And use something like Fixes ...[for autoclosing](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) the issue.

**Review by calixtus — APPROVED** [review]
Looks good to me too, one thing that could be improved in a future PR is making the executable binding dependant on if there is a doi present.

**Comment by calixtus** [comment]
Thanks for your contribution to JabRef!

**Pull request merged**